### PR TITLE
[12.0][ADD] purchase_deposit

### PR DESCRIPTION
**Pull request by kittiu**
Done
- [X] code review
- [X] readme files
- [X] test script

This module add ability to create purchase deposit, similar to sales deposit.

### CODE REVIEW

**Comment by kittiu** [comment]
Currency there is following error,
```
************* Module purchase_deposit
purchase_deposit/__init__.py:1: [C7902(missing-readme), ] Missing ./README.rst file. Template here: https://github.com/OCA/maintainer-tools/blob/master/template/module/README.rst
************* Module purchase_exception
purchase_exception/readme/USAGE.rst:1: [W7908(missing-newline-extrafiles), ]  Missing newline
```

1st error of missing README.rst file, I think it is not critical, right? As it will be created by Bot after merged (do I understand correctly?)

2nd error is from other module purchase_exception. When this happen, can I fix it in this PR? @pedrobaeza

File: purchase_deposit/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Copyright 2019 Elico Corp, Dominique K. <dominique.k@elico-corp.com.sg>
+# Copyright 2019 Ecosoft Co., Ltd., Kitti U. <kittiu@ecosoft.co.th>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+{
+    'name': 'Purchase Deposit',
+    'version': '12.0.0.0.0',
```
Comment (nikul-serpentcs) [inline purchase_deposit/__manifest__.py]: @kittiu `12.0.1.0.0`

**Review by nikul-serpentcs — COMMENTED** [review]
Minor Change

File: purchase_deposit/__manifest__.py (hunk 1 above)
Comment (kittiu) [reply purchase_deposit/__manifest__.py]: This is new module not migration, do I we need .1?

Comment (kittiu) [reply purchase_deposit/__manifest__.py]: Yes, it need :p. Fixed.

File: purchase_deposit/wizard/purchase_make_invoice_advance.py [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+
+    @api.model
+    def view_init(self, fields):
+        active_id = self._context.get('active_id')
+        purchase = self.env['purchase.order'].browse(active_id)
+        if purchase.state != 'purchase':
+            raise UserError(
+                _('This action is allowed only in Purchase Order sate'))
+        return super().view_init(fields)
+
```
Comment (nikul-serpentcs) [inline purchase_deposit/wizard/purchase_make_invoice_advance.py:52]: Remove One Line

**Review by nikul-serpentcs — COMMENTED** [review]
@kittiu Minor Change Because `Mr.Travis` Need more Love :heart:

File: purchase_deposit/wizard/purchase_make_invoice_advance.py (hunk 2 above)
Comment (kittiu) [reply purchase_deposit/wizard/purchase_make_invoice_advance.py:52]: @nikul-serpentcs Thanks!, fixed.

**Comment by kittiu** [comment]
@OCA/procurement-purchase-maintainers can this be merged :) ?

File: purchase_deposit/models/purchase.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+             for line in self.order_line.filtered(lambda l: not l.is_deposit)]
+        return super(PurchaseOrder, self).copy_data(default)
+
+
+class PurchaseOrderLine(models.Model):
+    _inherit = 'purchase.order.line'
+
+    is_deposit = fields.Boolean(
+        string='Is a deposit payment',
+        help="Deposit payments are made when creating invoices from a purhcase"
```
Comment (HviorForgeFlow) [inline purchase_deposit/models/purchase.py:26]: typo: purhcase

File: purchase_deposit/wizard/purchase_make_invoice_advance.py [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+        string='What do you want to invoice?',
+        default='percentage',
+        required=True,
+    )
+    purchase_deposit_product_id = fields.Many2one(
+        comodel_name='product.product',
+        string='Deposit Payment Product',
+        domain=[('type', '=', 'service')],
+    )
+    count = fields.Integer(
```
Comment (HviorForgeFlow) [inline purchase_deposit/wizard/purchase_make_invoice_advance.py]: Which is the usage of this field. I can figure out that could be used to create a deposit for multiple PO's, but I can't reproduce it.

**Review by HviorForgeFlow — CHANGES_REQUESTED** [review]
Code & Functional review

File: purchase_deposit/wizard/purchase_make_invoice_advance.py (hunk 4 above)
Comment (kittiu) [reply purchase_deposit/wizard/purchase_make_invoice_advance.py]: Yes, I wasn't needed. This part try to be similar to SO, but you are right, there is no such action in PO. So, I removed it.
Thank you @hveficent

**Comment by kittiu** [comment]
@nikul-serpentcs all fixed. Would you help approve this? Thx!

**Review by HviorForgeFlow — APPROVED** [review]

**Review by ps-tubtim — APPROVED** [review]

**Pull request merged**